iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -146,20 +148,33 @@ class Sample
     ///
     const T* get() const noexcept;
 
+    using ConditionalConstChunkHeader_t = typename cxx::add_const_conditionally<mepoo::ChunkHeader, T>::type;
```

Comment:
Could you please explain in a short doxygen comment why we need a `ConditionalConstChunkHeader`?